internal/service/domain/treecluster/handle_new_sensor_data.go

```diff
@@ -31,78 +31,100 @@ func (s *TreeClusterService) HandleNewSensorData(ctx context.Context, event *ent
 		return nil
 	}
 
-	sensorData, err := s.treeClusterRepo.GetAllLatestSensorDataByClusterID(ctx, tree.TreeCluster.ID)
+	wateringStatus, err := s.getWateringStatusOfTreeCluster(ctx, tree.TreeCluster.ID)
 	if err != nil {
-		log.Error("failed to get latest sensor data", "cluster_id", tree.TreeCluster.ID, "error", err)
 		return nil
 	}
```

Comment:
Inside the `getWateringStatusOfTreeCluster` function you return lots of custom errors but do nothing with them? Maybe log the error?

(It's good that you don't return the error because the event manager will stop his work when it receives an error :+1:)